imnodes.cpp

```diff
@@ -1108,8 +1112,10 @@ void begin_canvas_interaction(EditorContext& editor)
             ? (g.left_mouse_clicked && *g.io.emulate_three_button_mouse.modifier)
             : g.middle_mouse_clicked;
 
-    editor.click_interaction_type =
-        started_panning ? ClickInteractionType_Panning : ClickInteractionType_BoxSelection;
+    editor.click_interaction_type = started_panning
+                                        ? ClickInteractionType_Panning
+                                        : g.left_mouse_clicked ? ClickInteractionType_BoxSelection
+                                                               : editor.click_interaction_type;
```

Comment:
You've made the code much cleaner in other parts already -- and I think this could also be improved by doing away with the nested conditionals and replaced with an `if`. That way, we could also get rid of the additional `if` two lines below. So something like:

```cpp
    if (started_panning)
    {
        editor.click_interaction_type = ClickInteractionType_Panning;
    }
    else if (g.left_mouse_clicked)
    {
        editor.click_interaction_type = ClickInteractionType_BoxSelection;
        editor.click_interaction_state.box_selector.rect.Min = g.mouse_pos;
    }
```